server/internal/router/router.go

```diff
@@ -0,0 +1,184 @@
+package router
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	v1 "github.com/llm-operator/inference-manager/api/v1"
+	"github.com/llm-operator/inference-manager/server/internal/config"
+	"github.com/llm-operator/inference-manager/server/internal/k8s"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"k8s.io/apimachinery/pkg/util/runtime"
+)
+
+const tickPeriod = 60 * time.Second
+
+type model struct {
+	id string
+}
+
+// Router manages the routing table of models to engines.
+type Router struct {
+	// m is the route map.
+	m *route
+
+	imeConfig config.InferenceManagerEngineConfig
+}
+
+// NewRouter creates a new router.
+func NewRouter(c config.InferenceManagerEngineConfig) *Router {
+	return &Router{
+		m:         newRoute(),
+		imeConfig: c,
+	}
+}
+
+// Run updates the routes periodically and upon pod events.
+func (r *Router) Run(ctx context.Context, errCh chan error) error {
+	k8sClient, err := k8s.NewClient()
```

Comment:
I think we want to initialize the k8s client or informer outside of this class. Then we can inject a fake k8s client.